src/storage/tests.rs

```diff
@@ -295,6 +295,39 @@ mod test {
         }
     }
 
+    fn test_get_mut_or_default<T: Component + Default + From<u32> + AsMut<u32> + Debug + Eq>()
+    where
+        T::Storage: Default,
+    {
+        let mut w = World::new();
+        let mut s: Storage<T, _> = create(&mut w);
+
+        for i in 0..500 {
+            if let Err(err) = s.insert(Entity::new(i, Generation::new(1)), (i + 1290).into()) {
+                panic!("Failed to insert component into entity! {:?}", err);
+            }
+        }
+
+        for i in 0..1_000 {
```

Comment:
hm, what are these loops and magic numbers? am finding I have to think quite a bit to understand the logic, and then think again to figure out what its intent is.

if the loops are necessary, it's worth writing the rationale in a comment
* is it to get the entity index high enough?
* can we use a smaller backing storage to achieve the same effect?

not obvious to understand what the test is doing 🤔:

1. insert 500 entities with `n + 1290` as its value.
2. mutate 1000 entities with `get_mut_or_default()` to be `n + n + 710`
3. assert the first 500 are `2n + 2000`
4. assert the last 500 are `n + 710`

ah 💡! while it is logically correct and does test the effect, it's considerable mental effort to turn the code back into understanding. could you try to make it easier, otherwise my head is like this: 🤕, haha

---

**edit:** just expanded the diff, and saw the *existing code* is like that. oh man 😂

Reply:
I removed the magic numbers & commented it.